### PR TITLE
fix(ivy): support for #id bootstrap selectors

### LINKED ISSUE #33485: bootstrapping apps with `selector: '#app'` in `app.component.ts` is not working any more after upgrading to 9.0.0
# 🐞 bug report

### Affected Package

 The issue is caused by package `@angular/core` or `@angular/compiler` ? I don't know exactly.

### Is this a regression?

 Yes, the previous version in which this bug was not present was: ....

Yes, in `8.0.0`, it does work.

### Description

 A clear and concise description of the problem...

As title, when I'm using `#app`, it complains about `selctor 'app' not found`.

## 🔬 Minimal Reproduction

 https://stackblitz.com/...

## 🔥 Exception or Error
<pre><code>

</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>

</code></pre>

**Anything else relevant?**

### PR DESCRIPTION
**Pull request by mhevery**
Fixes: #33485

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:280309077:BASE:280309129:1573691947786:284cb44)
[ivy presubmit](http://test/OCL:280309077:BASE:280309674:1573692106468:8ed9986b)

File: packages/core/test/acceptance/bootstrap_spec.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+import {Component, NgModule} from '@angular/core';
+import {getComponentDef} from '@angular/core/src/render3/definition';
+import {BrowserModule} from '@angular/platform-browser';
+import {platformBrowserDynamic} from '@angular/platform-browser-dynamic';
+import {onlyInIvy, withBody} from '@angular/private/testing';
+
+describe('bootstrap', () => {
+  onlyInIvy('VE does not allow reading selectors').it('Component should have id selector', () => {
```
Comment (pkozlowski-opensource) [inline packages/core/test/acceptance/bootstrap_spec.ts]: nit: personally I would drop this test for the following reason:
* we are picking into "implementation details" here;
* we've got a compiler test for a selector parsing already in https://github.com/angular/angular/pull/33784/files#diff-ee14363ac9344a65ec471398a50b0ca2R678

Tests are good but I'm worried to the maintenance cost here.

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM with one nit

File: packages/compiler-cli/test/compliance/r3_compiler_compliance_spec.ts [hunk 2]
```diff
@@ -675,6 +675,35 @@ describe('compiler compliance', () => {
       expectEmit(source, OtherDirectiveFactory, 'Incorrect OtherDirective.ɵfac');
     });
 
+    it('should convert #myApp selector to ["", "id", "myApp"]', () => {
```
Comment (kara) [inline packages/compiler-cli/test/compliance/r3_compiler_compliance_spec.ts]: Test description should match test

```suggestion
    it('should convert #my-app selector to ["", "id", "my-app"]', () => {
```

File: packages/compiler/src/selector.ts [hunk 3]
```diff
@@ -57,28 +71,37 @@ export class CssSelector {
     let inNot = false;
     _SELECTOR_REGEXP.lastIndex = 0;
     while (match = _SELECTOR_REGEXP.exec(selector)) {
-      if (match[1]) {
+      if (match[SelectorRegexp.NOT]) {
```
Comment (kara) [inline packages/compiler/src/selector.ts]: 👍 Thanks for cleaning this up

**Review by kara — APPROVED** [review]
LGTM, one nit

File: packages/core/test/acceptance/bootstrap_spec.ts (hunk 1 above)
Comment (mhevery) [reply packages/core/test/acceptance/bootstrap_spec.ts]: dropped.

**Pull request closed without merging**